### ISSUE TITLE
Storing and Loading document attachments

### ISSUE THREAD

**Issue by andrewcstewart** [ropensci/sofa#26]
I've been playing around with how to do this within R and was wondering what your thoughts are.  I see you a getattach and doc_attach functions, though doc_attach isn't implemented yet and I can't get getattach to work for me.  I get the following error, if curious:

```
getattach(dbname = "mydb", docid = "a99569a30658181ff9e3e7340d00e2fc", attname = "foobar.Rds")
Error in parseJSON(txt) : lexical error: invalid char in json text.
                                       
                     (right here) ------^
```

I've also been trying to directly fetch an attachment from a couchdb url into my current R session (an RDS file) without any luck.

```
readRDS(url("http://atlas.veracyte.lan:5984/uci/a99569a30658181ff9e3e7340d00e2fc/mag7.eset.rds",open = "rb"))
Error: unknown input format
```

Wondering if you have any insight on either of those two cases.

**Comment by sckott** [comment]
@andrewcstewart 

for the `getattach()` function, I haven't worked on that much - I'll take a longer look at it. It probably needs to be smart about detecting the file type and if binary writing to disk and if character/json/xml, then could read it in to memory if below a certain file size

i imagine the `.rds` file being a binary file would take a little more finess, but essentially we could use e.g., 

```r
GET(url, write_disk(path="...", overwrite=TRUE))
```

**Comment by sckott** [comment]
@andrewcstewart i'm reworking these functions a bit, and renaming them following with other fxn name changes

**Comment by sckott** [comment]
closed via 6edc42113bc78d2994891e5be2d42fa9d8d78565
